core/src/main/scala/org/locationtech/rasterframes/extensions/RasterJoin.scala

```diff
@@ -36,7 +36,7 @@ import scala.util.Random
 object RasterJoin {
```

Comment:
I'm torn on what to do here with the string-ly typed resample methods. It's what I suggested as a hack around the  serialization error, but ultimately I'd want the Scala API to use first-class types as must as possible in the user-facing API. Even if that mean turning it into a string to instantiate the `Expression`. I don't want to add a bunch of onerous work, but I suspect I'll regret the public API not using strong type. We can leave it as is to get it out the door, then change it later as a breaking change in Scala (Python would remain string-ly), or make some extra effort here to get the Scala-land API ship shape. I'll let you decide.

Reply:
ah ... in my path dependent thinking i did not even really think through having it typed in the scala api... much easier tbh